File: javascript/Contagem.js
function contarValoresInteiros(conjuntoDeDados, N) {
  // Inicializa a contagem
  let contagem = 0;

  // Percorre o conjunto de dados
  for (let i = 0; i < conjuntoDeDados.length; i++) {
    // Verifica se o valor atual é um número inteiro entre 1 e N
    if (Number.isInteger(conjuntoDeDados[i]) && conjuntoDeDados[i] >= 1 && conjuntoDeDados[i] <= N) {
      contagem++;
    }
  }

  // Retorna o resultado da contagem
  return contagem;
}

// Exemplo de uso
const conjuntoDeDados = [3, 5, 2, 7, 1, 4, 6];
const N = 5;

const resultado = contarValoresInteiros(conjuntoDeDados, N);
console.log(`A contagem de valores inteiros entre 1 e ${N} é: ${resultado}`);

File: java/Contagem.java
import java.util.Scanner;

public class Contagem {

	public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        System.out.print("Informe um número N: ");
        int N = scanner.nextInt(); 

        int[] dd = {1, 3, 5, 7, 9, 12, 15, 18, 20, 22, 25, 27, 30};
        int cn = 0;

        for (int i = 0; i < dd.length; i++) {
            if (dd[i] >= 1 && dd[i] <= N) { 
                cn++;
            }
        }

        System.out.println("Existem " + cn + " valores inteiros entre o primeiro dado e " + N);
    }

}
